go-controller/pkg/ovn/policy.go

```diff
@@ -104,17 +106,22 @@ func (oc *Controller) syncNetworkPolicies(networkPolicies []interface{}) {
 	}
 }
 
-func addAllowACLFromNode(logicalSwitch string, mgmtPortIP net.IP) error {
+func addAllowACLFromNode(logicalSwitch string, mgmtPortIP net.IP, ovnNBClient goovn.Client) error {
 	ipFamily := "ip4"
 	if utilnet.IsIPv6(mgmtPortIP) {
 		ipFamily = "ip6"
 	}
 	match := fmt.Sprintf("%s.src==%s", ipFamily, mgmtPortIP.String())
-	_, stderr, err := util.RunOVNNbctl("--may-exist", "acl-add", logicalSwitch,
-		"to-lport", defaultAllowPriority, match, "allow-related")
-	if err != nil {
-		return fmt.Errorf("failed to create the node acl for "+
-			"logical_switch=%s, stderr: %q (%v)", logicalSwitch, stderr, err)
+
+	priority, _ := strconv.Atoi(defaultAllowPriority)
+	aclcmd, err := ovnNBClient.ACLAdd(logicalSwitch, "to-lport", match, "allow-related", priority, nil, false, "", "")
+	if err == nil {
+		// NOTE: goovn.ErrorExist is returned if the ACL already exists, skip returning an error
+		if err = ovnNBClient.Execute(aclcmd); err != nil && err != goovn.ErrorExist {
```

Comment:
If the ACL already exists, you'll get the "ErrorExist" error from the ACLAdd call.  And, if there's an error, the aclcmd returned is nil.  If you then call Execute with a nil command, it's treated as a no-op and will execute without error.

I think you need to move the check for ErrorExist above the check for "err == nil".

Reply:
@anfredette good catch, Thanks